src/main/java/com/google/sps/servlets/AddToDatabaseServlet.java

```diff
@@ -0,0 +1,82 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.QuestionChoice;
+import com.google.sps.data.QuestionDatabase;
+import com.google.sps.data.QuizQuestion;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.PatternSyntaxException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/addToDatabase")
+/** Responds with a JSON string containing questions and answers for the career quiz */
+public class AddToDatabaseServlet extends HttpServlet {
+  private static final String JSON_CONTENT_TYPE = "application/json";
+  private static final String CAREER_QUIZ_FORM_SUBMIT_PARAMETER = "careerQuizSubmit";
+  private static final String CAREER_QUIZ_QUESTION_PARAMETER = "careerQuestion";
+  private static final String CAREER_QUIZ_CHOICETEXT_PARAMETER = "careerChoiceText";
+  private static final String CAREER_QUIZ_CAREERPATH_PARAMETER = "careerPath";
+  private static final String CAREER_DATABASE_ENTITY_QUERY_STRING = "careerQuizQuestionAndChoices";
+  private static final String CANT_SPLIT_STRING =
+      "Choices and career paths need to be split by semicolons.";
+  private static final String NUMBER_CHOICES_DONT_MATCH_CAREER_PATHS =
+      "The number of choices given is not the same as the number of career paths given.";
+  private static final String REDIRECTION_URL = "admin/AddStaticData.html";
+  private static final Gson gson = new Gson();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Should this be created in a constructor? Then you can pass in the datastore in your test instead of hoping that the instance created by the factory is the same both times.

Reply:
A servlet's constructor will never be invoked upon initialization (https://stackoverflow.com/questions/1276082/how-to-initialize-a-servlet-during-startup-with-parameters), which is why I tried to use Guice previously but we decided not to go that route.